File: 2025-06-June-GFG-POTD/June 19 - Case-specific Sorting of Strings/Solution.py
class Solution:
    @staticmethod
    def caseSort(s: str) -> str:
        upper = []
        lower = []

        # Separate characters by case
        for c in s:
            if c.isupper():
                upper.append(c)
            else:
                lower.append(c)

        # Sort both lists
        upper.sort()
        lower.sort()

        # Reconstruct the result
        result = []
        i = j = 0

        for c in s:
            if c.isupper():
                result.append(upper[i])
                i += 1
            else:
                result.append(lower[j])
                j += 1

        return ''.join(result)

File: 2025-06-June-GFG-POTD/June 19 - Case-specific Sorting of Strings/Solution.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    string caseSort(string s) {
        vector<char> upper, lower;

        // Separate characters by case
        for (char c : s) {
            if (isupper(c))
                upper.push_back(c);
            else
                lower.push_back(c);
        }

        // Sort both vectors
        sort(upper.begin(), upper.end());
        sort(lower.begin(), lower.end());

        // Reconstruct the result string
        string result = "";
        int i = 0, j = 0;

        for (char c : s) {
            if (isupper(c)) {
                result += upper[i++];
            } else {
                result += lower[j++];
            }
        }

        return result;
    }
};
